feats/app.py

```diff
@@ -43,6 +42,19 @@ def create(self, *args) -> object:
         impl_name = self.find(*args)
         return self.feature.implementations[impl_name].fn(*args)
 
+    def set_state(self, new_state: FeatureState):
+        serialized_state = copy.deepcopy(new_state.serialize(self.app))
+        self.app.storage[self.name].append(serialized_state)
+
+    def get_current_state(self) -> FeatureState:
+        states = self.app.storage[self.name]
+        try:
+            state_data = copy.deepcopy(states[-1])
```

Comment:
Whats the driver behind using `deepcopy` in these two state methods?

Reply:
Without a deep copy, it is possible to modify the state without going through the set_state interface. This caused issues with the memory store & testing.

Reply:
That makes sense 👍